### ISSUE TITLE
The UI should not allow node pools with the same name

### ISSUE THREAD

**Issue by sowmyav27** [rancher/dashboard#4444]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**
- Deploy a node driver DO cluster. - Two nodepools with same name
```
sowmya-cis-1 --> etcd/control plane - 1 node
sowmya-cis-1 --> worker role - 1 node
```
- worker node is seen stuck in registering state.
- Scale up the worker node - `sowmya-cis-3` is added.
<img width="1014" alt="Screen Shot 2021-04-22 at 4 05 06 PM" src="https://user-images.githubusercontent.com/26032343/115796842-65470e00-a387-11eb-8eb4-054ce3c30fc3.png">

**Other details that may be helpful:**

- This is not happening always.
- Reach out t me for rancher server logs

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.5.8-rc7
- Installation option (single install/HA): HA AKS local cluster

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): RKE DO node driver
- Kubernetes version (use `kubectl version`):

```
1.20 (Any)
```

**Comment by nwmac** [comment]
We should not allow node pools with the same name - note this is in the Ember UI

**Comment by catherineluse** [comment]
Moving to test because a fix was made for this issue here in this PR: https://github.com/rancher/ui/pull/4842

**Comment by gaktive** [comment]
For QA, this looks to have been addressed in a fix made for 2.6.7 so this hopefully is a simple validation.
Related tickets:
- https://github.com/rancher/dashboard/issues/4465
- https://github.com/rancher/dashboard/issues/4444
